dan_layer/engine/src/runtime/impl.rs

```diff
@@ -282,18 +294,22 @@ impl<TTemplateProvider: TemplateProvider<Template = LoadedTemplate>> RuntimeInte
 
         self.invoke_modules_on_runtime_call("emit_event")?;
 
-        let component_address = self.tracker.read_with(|state| {
+        let component_address_option = self.tracker.read_with(|state| {
             Ok::<_, RuntimeError>(
                 state
                     .current_call_scope()?
                     .get_current_component_lock()
                     .and_then(|l| l.address().as_component_address()),
             )
         })?;
+        let substate_id = match component_address_option {
+            Some(component_address) => Some(SubstateId::Component(component_address)),
+            None => None,
+        };
```

Comment:
https://rust-lang.github.io/rust-clippy/master/index.html#/manual_map

```suggestion
        let substate_id = component_address_option.map(|addr| SubstateId::Component(addr));
```